### PR TITLE
add linting support

### PR DESCRIPTION
**Pull request by jaymccon**
initial stab at adding linting support to taskcat.

By default only warns, to prevent changing any existing behaviors. set `-l error` to fail on errors and `-l strict` to fail on warnings as well.

### CODE REVIEW

File: taskcat/stacker.py [hunk 1]
```diff
@@ -1722,8 +1722,92 @@ def check_cfnyaml(self, yamlin, quiet=None, strict=None):
...
+        :return: set of nested template paths
+        """
+        children = set()
+        template = cfnlint.decode.cfn_yaml.load(filename)
+        for resource in template['Resources'].keys():
+            child_name = ''
+            if template['Resources'][resource]['Type'] == "AWS::CloudFormation::Stack":
+                template_url = template['Resources'][resource]['Properties']['TemplateURL']
+                if type(template_url) == dict:
+                    if 'Fn::Sub' in template_url.keys():
```
Comment (agamzn) [inline taskcat/stacker.py]: You're only accounting for 'Fn::Sub', not '!Sub'.

File: taskcat/stacker.py [hunk 2]
```diff
@@ -1722,8 +1722,92 @@ def check_cfnyaml(self, yamlin, quiet=None, strict=None):
...
+                    t,
+                    template,
+                    rules,
+                    lints[test]['regions'])
+        test_status = {"W": False, "E": False}
+        for test in lints.keys():
+            for t in lints[test]['results'].keys():
+                print(I + "Lint results for test %s on template %s:" % (test, t))
+                for r in lints[test]['results'][t]:
+                    message = r.__str__().lstrip('[')
```
[the remark this answers is not in the document]
Comment (jaymccon) [reply taskcat/stacker.py]: it's not, it's a custom object created by the linting class (that doesn't implement lstrip)

File: taskcat/stacker.py (hunk 1 above)
Comment (jaymccon) [reply taskcat/stacker.py]: the (awesome) linting library that we're now using to parse templates converts all shorthand for us. so !Sub becomes Fn::Sub by the time this code gets to it

File: taskcat.py [hunk 3]
```diff
@@ -42,6 +43,10 @@ def main():
             taskcat_cfg = yaml.safe_load(cfg.read())
         cfg.close()
 
+        project_path = '/'.join(tcat_instance.get_config().split('/')[0:-2])
+        os.chdir(os.path.abspath(project_path))
+        tcat_instance.lint(args.lint)
+        os.chdir(os.path.abspath('../'))
```
[the remark this answers is not in the document]
Comment (jaymccon) [reply taskcat.py]: will implement

File: taskcat/stacker.py [hunk 4]
```diff
@@ -1722,8 +1722,92 @@ def check_cfnyaml(self, yamlin, quiet=None, strict=None):
...
+        rules = cfnlint.core.get_rules([], [])
+        for test in config['tests'].keys():
+            lints[test] = {}
+            if 'regions' in config['tests'][test].keys():
+                lints[test]['regions'] = config['tests'][test]['regions']
+            else:
+                lints[test]['regions'] = config['global']['regions']
+            template_file = config['tests'][test]['template_file']
+            template_file = 'templates/' + template_file
+            lints[test]['template_file'] = template_file
```
[the remark this answers is not in the document]
Comment (jaymccon) [reply taskcat/stacker.py]: will implement

File: taskcat/stacker.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (agamzn) [reply taskcat/stacker.py]: 👍

File: taskcat/stacker.py (hunk 1 above)
Comment (agamzn) [reply taskcat/stacker.py]: 👍

**Comment by agamzn** [comment]
Ship it!

**Pull request merged**